frontend/appflowy_flutter/lib/plugins/database/grid/presentation/widgets/filter/filter_menu.dart

```diff
@@ -35,7 +35,12 @@ class FilterMenu extends StatelessWidget {
           final List<Widget> children = [];
           children.addAll(
             state.filters
-                .map((filterInfo) => FilterMenuItem(filterInfo: filterInfo))
+                .map(
+                  (filterInfo) => FilterMenuItem(
+                    key: ValueKey(filterInfo.filter.id),
```

Comment:
Resolves issue where deleting filters of same name and field type only works once, then the rest are stuck